src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -308,12 +309,14 @@ public static double log10sumLog10(final double[] log10p, final int start, final
 
 
     /**
-     * A helper class to maintain a cache of log values.
-     * The cache is immutable after creation.
+     * A helper class to maintain a cache of log10 values.
+     * The cache expands when a number is not available.
+     * NOTE: this cache is NOT thread safe and it must not be accessed from multiple threads.
      */
     private static final class Log10Cache {
```

Comment:
An instance of this cache is you used to back the static method `MathUtils.log10`.  It was previously effectively immutable so it wasn't a problem to have a static cache. Now that this is mutable though I think you should probably guard those instances in ThreadLocals so that it won't be caught in some broken state if multiple threads call `MathUtils.log10`